src/Components/src/Microsoft.AspNetCore.Components/IComponent.cs

```diff
@@ -12,12 +14,12 @@ public interface IComponent
         /// Initializes the component.
         /// </summary>
         /// <param name="renderHandle">A <see cref="RenderHandle"/> that allows the component to be rendered.</param>
-        void Init(RenderHandle renderHandle);
+        void Configure(RenderHandle renderHandle);
```

Comment:
Has a lot of thought gone into this name? I think this is a pretty strange choice.

Reply:
@SteveSandersonMS  has thoroughly thought of it. What is that you don't like about it? I like better than Init, that's for sure, as it at least doesn't clash with OnInit and OnInitAsync

Reply:
I think Init or Initialize is a wholly more appropriate name. Naming things like `Initialize`, `OnInitalize`, `OnInitializeAsync` makes it obvious that they are related.

Reply:
But they are not related, that's the point.

OnInit and OnInitAsync run on the first SetParametersAsync only and are a convenience from componentbase.

Init (now Configure) is just there so that we can pass in the renderhandle

Reply:
Yep. The `IComponent.Init` method name was confusing because it seems like it's the same thing as `ComponentBase.Init`, but really it's totally different. `IComponent.Init` isn't really a lifecycle method; it's just a way of supplying some infrastructure to the component (like you'd do in a constructor if interfaces could describe constructors).

As such I think it's great to remove some of this confusion by renaming `IComponent.Init` to `IComponent.Configure`.

Reply:
Ah, ok that makes more sense. I still don't like the naming but I don't have a better suggestion.

Reply:
`Setup` would also be fine, and maybe communicates the "one time"-ness of it better, but I'm OK with either. Opinions?

Reply:
`Attach`?

I don't want to spend too much time debating this right now since we'll review all of this API surface all-up eventually.

Reply:
OK, I'd vote with "leave it as `Configure` until we review" to avoid more debate and disruption then :)